core/src/main/java/org/keycloak/representations/idm/CredentialRepresentation.java

```diff
@@ -155,4 +155,101 @@ public void setCreatedDate(Long createdDate) {
     public void setConfig(MultivaluedHashMap<String, String> config) {
         this.config = config;
     }
+
```

Comment:
The UserTest doesn't pass without it. The assertAdminEvents.assertEvent() in UserTest.createUser(UserRepresentation) checks the stored admin event, which can include the credentialrepresentation. It failed without the implemented equals method.